src/libraries/System.ServiceProcess.ServiceController/tests/ServiceControllerTests.netcoreapp.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.ServiceProcess.Tests
+{
+    [OuterLoop(/* Modifies machine state */)]
+    public partial class ServiceControllerTests : IDisposable
+    {
+        [ConditionalFact(nameof(IsProcessElevated))]
+        public void Stop_FalseArg_WithDependentServices_ThrowsInvalidOperationException()
+        {
+            var controller = new ServiceController(_testService.TestServiceName);
+            controller.WaitForStatus(ServiceControllerStatus.Running, _testService.ControlTimeout);
+            Assert.Throws<InvalidOperationException>(() => controller.Stop(stopDependentServices: false));
+        }
+
+        [ConditionalFact(nameof(IsProcessElevated))]
+        public void Stop_TrueArg_WithDependentServices_StopsTheServiceAndItsDependents()
+        {
+            var controller = new ServiceController(_testService.TestServiceName);
+            controller.WaitForStatus(ServiceControllerStatus.Running, _testService.ControlTimeout);
+
+            controller.Stop(stopDependentServices: true);
```

Comment:
We should also add a test that `controller.Stop(false)` works correctly. This one really only tests the same thing as just calling `Stop()`.

From reading the issue, the reason we are adding this new API is so someone can call `controller.Stop(false)`. So we should be testing it.

Reply:
You are right. I will try to create another test that acts on a test service without dependents.

Reply:
> without dependents

I think it would be best if the test ensured the scenario listed in https://github.com/dotnet/runtime/issues/35284 worked correctly.

Reply:
Ok, I think I got what you meant. You'd like to see a test that manually stops all the dependent services and then the service itself with Stop(false), right?

Reply:
Right - that was the motivating reason for adding this API, so it would be good to ensure it works.

Reply:
Added a test for that scenario as you requested, tell me if it's ok as-is.

Reply:
Perfect! That's exactly what I was looking for. Thanks!